specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-10-01-preview/BulkActions.json

```diff
@@ -0,0 +1,652 @@
+                  "value": "AddToSet"
+                },
+                {
+                  "description": "Use this operator to add an element to the end of a list (i.e. an ordered multivalue variable).",
+                  "value": "AppendToList"
+                },
+                {
+                  "description": "Use this operator to remove the specified element from a set (i.e. an unordered multivalue variable).",
+                  "value": "RemoveFromSet"
+                }
+              ]
+            }
+          },
+          "value": {
+            "description": "Value to be applied by this operation step.",
+            "type": "string"
+          }
+        }    
+      },
+      "Operation": {
+        "description": "Represents the operation to be applied to all items captured by this bulk action.",
+        "type": "object",
+        "properties": {
+          "type": {
+            "description": "Describes which type of operation we are applying to items captured by this bulk action.",
+            "type": "string",
+            "enum": [
+              "Delete",
+              "Edit"
+            ],
+            "x-ms-enum": {
+              "modelAsString": true,
+              "name": "OperationType",
+              "values": [
+                {
+                  "description": "This item will be deleted.",
+                  "value": "Delete"
+                },
+                {
+                  "description": "This item will be edited.",
+                  "value": "Edit"
+                }
+              ]
+            }
+          },
+          "steps": {
+            "description": "List of operation steps to be applied.",
+            "type": "array",
+            "items": {
+              "type": "object",
+              "$ref": "#/definitions/OperationStep"
+            },
+            "x-ms-identifiers": []
+          }
+        }
+      },
+      "BulkAction": {
+        "allOf": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Resource"
+          }
+        ],
+        "description": "Represents a bulk action in Azure Security Insights.",
+        "properties": {
+          "properties": {
+            "$ref": "#/definitions/BulkActionProperties",
+            "description": "Bulk action properties",
+            "x-ms-client-flatten": true
+          }
+        },
+        "type": "object"
+      },
+      "BulkActionList": {
+        "description": "List all the bulk actions.",
+        "properties": {
+          "nextLink": {
+            "description": "URL to fetch the next set of bulk actions.",
+            "readOnly": true,
+            "type": "string"
+          },
+          "value": {
+            "description": "Array of bulk actions.",
+            "items": {
+              "$ref": "#/definitions/BulkAction"
+            },
+            "type": "array",
+            "x-ms-identifiers": []
```

Comment:
>[] [](http://example.com/codeflow?start=32&length=2)

The id property uniquely identifies a BulkAction so you can place the id property here

Reply:
Apply this everywhere. If there are properties that uniquely identify an element in the array, specify that here. You can specify more than one property that uniquely identifies an element.